src/decoder.rs

```diff
@@ -8,34 +8,56 @@ use nom::bytes::complete::{take, take_till};
 use nom::combinator::{map, map_parser};
 use nom::multi::many0;
 use nom::number::complete::{be_f32, be_f64, be_i32, be_i64, be_u32};
-use nom::{IResult,combinator::map_res,sequence::tuple};
+use nom::{IResult,combinator::map_res,sequence::tuple,Err};
+use OscError::BadPacket;
 
 /// Common MTU size for ethernet
 pub const MTU: usize = 1536;
 
 /// Takes a bytes slice representing a UDP packet and returns the OSC packet as well as a slice of
 /// any bytes remaining after the OSC packet.
-pub fn decode_udp<'a>(msg: &'a [u8]) -> IResult<&'a [u8], OscPacket, OscError> {
-    decode_packet(msg, msg)
+pub fn decode_udp<'a>(msg: &'a [u8]) -> Result<(&'a [u8], OscPacket), OscError> {
+    match decode_packet(msg, msg) {
+        Ok((remainder, osc_packet)) => Ok((remainder, osc_packet)),
+        Err(e) => match e {
+            Err::Incomplete(_) => Err(BadPacket("Incomplete data")),
+            Err::Error(e) => Err(e),
```

Comment:
Should we use `Err::Error(e) | Err::Failure(e) => Err(e),` to handle the non `nom::Error` cases?  I've actually no real preference, what do you think?